### PR TITLE
[networkx] implement (weighted) betweenness centrality in graphscope.

### PR DESCRIPTION
**Pull request by septicmk**
## What do these changes do?

implement (weighted) betweenness centrality in GraphScope.
- BetweenessCentrality for distributed mode (unweighted graph only)
- BetweenessCentralityGeneric for duplicated mode

- [x] implement C++ PIE app with analytical engine API
- [x] check the correctness with NetworkX
- [x] check the scalability and performance in large graph
- [x] add to python built-in algorithms and add related test case

## Performance
Betweenness Centrality
|  Dataset   | GraphScope(work_num=4)  | NetworkX |
|  ----  | ----  | ---- |
| ca-GrQc | 100.1 s | 162.3 s|
| email-Eu-core  |  27.3 s |  69.2 s |
| delaunay_n14 | 1094.2 s | 2808.4 s |
| p2p-Gnutella08 | 227.9 s | 427.9 s |
| soc-sign-bitcoinalpha | 92.4 s | 138.5 s |

## Related issue number

#523

### CODE REVIEW

**Comment by sighingnow** [comment]
Hi @septicmk could please try to squash your commits, rebase to main, and forcely push there? to avoid any unwanted changes in other part beyond the network module.

File: analytical_engine/apps/centrality/betweenness/betweenness_centrality_generic_context.h [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#ifndef ANALYTICAL_ENGINE_APPS_CENTRALITY_BETWEENNESS_BETWEENNESS_CENTRALITY_GENERIC_CONTEXT_H_
+#define ANALYTICAL_ENGINE_APPS_CENTRALITY_BETWEENNESS_BETWEENNESS_CENTRALITY_GENERIC_CONTEXT_H_
+
+#ifdef NETWORKX
```
Comment (acezen) [inline analytical_engine/apps/centrality/betweenness/betweenness_centrality_generic_context.h:19]: Do we  only can use the app in networkx mudule(or folly::dynamic)?  please double check  this

**Comment by yecol** [comment]
Waiting for the update on performance report in summary.

File: analytical_engine/apps/centrality/betweenness/betweenness_centrality_generic.h [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+#include "core/utils/app_utils.h"
+#include "core/worker/default_worker.h"
+
+#include "apps/centrality/betweenness/betweenness_centrality_generic_context.h"
+
+namespace gs {
+
+/**
+ * @brief Compute the betweenness centrality for vertices. The betweenness
+ * centrality fora vertex v is the fraction of vertices it is connected to.
```
Comment (acezen) [inline analytical_engine/apps/centrality/betweenness/betweenness_centrality_generic.h:41]: typo: fora -> for a

**Comment by septicmk** [comment]
## Performance
Betweenness Centrality
|  Dataset   | GraphScope(work_num=4)  | NetworkX |
|  ----  | ----  | ---- |
| ca-GrQc | 100.1 s | 162.3 s|
| email-Eu-core  |  27.3 s |  69.2 s |
| delaunay_n14 | 1094.2 s | 2808.4 s |
| p2p-Gnutella08 | 227.9 s | 427.9 s |
| soc-sign-bitcoinalpha | 92.4 s | 138.5 s |

**Review by yecol — APPROVED** [review]

**Pull request merged**